src/Messaging/MessageBuffer.php

```diff
@@ -158,9 +158,17 @@ public function onData($data) {
                 $payloadLenBytes = $payload_length === 126 ? 2 : 8;
                 $headerSize      += $payloadLenBytes;
                 $bytesToUpack    = substr($data, $frameStart + 2, $payloadLenBytes);
-                $payload_length  = $payload_length === 126
-                    ? unpack('n', $bytesToUpack)[1]
-                    : unpack('J', $bytesToUpack)[1];
+
+                if ($payload_length === 126){
+                    $payload_length = unpack('n', $bytesToUpack)[1];
+                } else {
+                    if (PHP_INT_SIZE == 4) { // if 32bits PHP
+                        $bytesToUpack = substr($bytesToUpack, 4); //Keep only 4 last bytes
+                        $payload_length = unpack('N', $bytesToUpack)[1];
```

Comment:
I like the approach! Additionally, we should make sure the first 4 bytes are all zeros. We may not be able to represent the actual number on a 32 bit system in this case, but we do not for certain that it would exceed 2 GiB, so it's safe to assume it's "too large". This probably also explains the failing test case as pointed out by @BadWolf42.